app/controllers/assets/link_previews/tracks_controller.rb

```diff
@@ -0,0 +1,147 @@
+class Assets::LinkPreviews::TracksController < ApplicationController
+  def show
+    track = Track.find(params[:id])
+
+    map = MiniMagick::Image.open(get_map_url(track))
+    map.contrast
+
+    convert = MiniMagick::Tool::Convert.new
+    convert.size '380x300'
+    convert.xc 'black'
+    convert.draw 'fill #ffffff roundRectangle 0,0 380,300 5,5'
+    convert << 'app/assets/images/canvas.png'
+    convert.call
+
+    card = MiniMagick::Image.open('app/assets/images/canvas.png')
+
+    logo = MiniMagick::Image.open('app/assets/images/logo260.png')
+    logo.resize '24x24'
+
+    result = card.composite(map) do |c|
+      c.compose 'Over'    # OverCompositeOp
+      c.geometry '+10+10' # copy second_image onto first_image from (20, 20)
+    end
+
+    result = result.composite(logo) do |c|
+      c.compose 'Over'    # OverCompositeOp
+      c.geometry '+334+216' # copy second_image onto first_image from (20, 20)
+    end
+
+    result.draw "text 10,270 'Скорость'"
+    result.draw "text 10,285 '375 км/ч'"
+
+    result.draw "text 80,270 'Расстояние'"
+    result.draw "text 80,285 '5975м'"
+
+    result.draw "text 170,270 'Время'"
+    result.draw "text 170,285 '143 с'"
+
+    send_file result.path, type: result.mime_type, disposition: 'inline'
+  end
+
+  private
+
+  def get_map_url(track)
+    endpoint = 'https://maps.googleapis.com/maps/api/staticmap'
+
+    points = get_points(track)
+    points.map do |point|
+      point['color'] = color(point['h_speed'])
+    end
+
+    partitions = split_by_collor(points)
+    paths = get_paths_from_partitions(partitions)
+
+    start_marker = "size:tiny%7Ccolor:0x002288%7C#{points.first['latitude']},#{points.first['longitude']}"
+
+    params = paths << %W[
+      markers=#{start_marker}
+      maptype=satellite
+      size=180x120
+      scale=2
+      key=#{ENV['MAPS_API_KEY']}
+    ]
+    params = params.join('&')
+    "#{endpoint}?#{params}"
+  end
+
+  def split_by_collor(points)
+    partitions = [[]]
+    last_color = points.first['color']
+
+    points.each do |point|
+      if point['color'] != last_color
+        partitions.last << point
+        last_color = point['color']
+        partitions << []
+      end
+      partitions.last << point
+    end
+
+    partitions
+  end
+
+  def get_paths_from_partitions(partitions)
+    paths = []
+
+    partitions.each do |part|
```

Comment:
`map` should be sufficient here in place of:
```
paths = []
paritions.each ...
paths
```
It will be:
```
parititions.map do |part|
  ...
  "path=#{path_style}#{path_points}"
end
```